gnovm/pkg/gnolang/store.go

```diff
@@ -476,10 +476,9 @@ func (ds *defaultStore) loadObjectSafe(oid ObjectID) Object {
 			debug.Printf("loadObjectSafe by oid: %v, type of oo: %v\n", oid, reflect.TypeOf(oo))
 		}
 
-		ds.alloc.Allocate(oo.GetShallowSize())
-		// Alloc values other than shallow value,
-		// RefValue, e.g. keep sync with copyValueWithRefs().
-		AllocExpanded(ds.alloc, oo)
+		ss := oo.GetShallowSize()
+		rs := getRefSize(oo)
+		ds.alloc.Allocate(ss + rs)
```

Comment:
while restoring object, should allocate its shallow size along with the size of refValues that associated to the object, (see `copyValuesWithRef` in realm.go).  
the restore and allocation process should be atomic, otherwise will potentially cause inconsistencies between allocation and GC counting.